packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -129,6 +129,7 @@ abstract class EndpointDispatch {
         sendByteDataAsRaw: connector.endpoint.sendByteDataAsRaw,
       );
     } on SerializableException catch (exception) {
+      await session.close();
```

Comment:
Suggestion: Move this session close to a finally section.

There seems to be a lot of paths to take into consideration here, I would suggest adding the `session.close()` to a `finally` section so that it is always done, regardless of how this code is modified.

Finally docs: https://dart.dev/language/error-handling#finally